File: objective-c/bob/Bob.m
#import "Bob.h"

@implementation Bob

// 메서드 구현
- (NSString *)responseToInput:(NSString *)input {
    // 입력에 따라 다른 응답 반환
    if ([input hasSuffix:@"?"]) {
        // 질문일 경우
        return @"Sure.";
    } else if ([input isEqualToString:[input uppercaseString]] && [input stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]].length > 0) {
        // 대문자로만 이루어지는 경우(소리치는 경우)
        return @"Whoa, chill out!";
    } else if ([input hasSuffix:@"?"] && [input isEqualToString:[input uppercaseString]] && [input stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]].length > 0) {
        // 질문을 소리치는 경우
        return @"Calm down, I know what I'm doing!";
    } else if ([[input stringByTrimmingCharactersInSet: [NSCharacterSet whitespaceCharacterSet]] length] == 0) {
        // 아무것도 없는 경우
        return @"Fine. Be that way!";
    } else {
        // 기타 경우
        return @"Whatever.";
    }
}

@end

File: python/bob/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()   # 입력받은 문자열의 양쪽 공백 제거
    if hey_bob.isupper() and hey_bob.endswith("?"): # 대문자로 질문하는 경우
        return "Calm down, I know what I'm doing!"
    elif hey_bob.isupper(): # 대문자로 말하는 경우
        return "Whoa, chill out!"
    elif hey_bob.endswith("?"): # 질문하는 경우
        return "Sure."
    elif hey_bob == "": # 아무 말도 하지 않는 경우
        return "Fine. Be that way!"
    else:
        return "Whatever."  # 위의 경우에 해당하지 않는 모든 경우